roles/haproxy-loadbalancer/templates/haproxy.cfg.j2

```diff
@@ -0,0 +1,120 @@
+    tune.ssl.default-dh-param 2048
+
+    # turn on stats unix socket
+    stats socket /var/lib/haproxy/stats
+
+#---------------------------------------------------------------------
+# common defaults that all the 'listen' and 'backend' sections will
+# use if not designated in their block
+#---------------------------------------------------------------------
+defaults
+    mode                    http
+    log                     global
+    option                  httplog
+    option                  dontlognull
+    option http-server-close
+    option                  redispatch
+    retries                 3
+    timeout http-request    10s
+    timeout queue           1m
+    timeout connect         10s
+    timeout client          60m
+    timeout server          60m
+    timeout http-keep-alive 10s
+    timeout check           10s
+    maxconn                 3000
+
+#---------------------------------------------------------------------
+# main frontend which proxys to the backends
+#---------------------------------------------------------------------
+frontend  localhost
+    mode http
+    bind *:443 ssl crt {{ tls_pem_path }}
+
+    # redirect all HTTP traffic to HTTPS
+    acl is_https ssl_fc
+    http-request redirect scheme https unless is_https
+
+    # keycloak
+    acl keycloak hdr(host) -i keycloak.cyverse.at
+
+    http-request add-header X-Forwarded-Proto https
+
+    # userportal backend
+    acl userportal hdr(host) -i user.cyverse.at
+
+
+    #redirect to HTTPS if ssl_fc is false / off.
+    redirect scheme https code 301 if !{ ssl_fc }
+    #redirect scheme https code 301
+    #http-request redirect scheme https if http
+
+    use_backend userportal if userportal
+    default_backend keycloak
+
+
+#---------------------------------------------------------------------
+# round robin balancing between the Keycloak backends
+#---------------------------------------------------------------------
+backend keycloak
+    balance roundrobin
+    option  log-health-checks
+    option  forwardfor
+    # TODO server sX
+{% for host in groups['k8s-workers'] %}
+    server   {{ hostvars[host].ansible_hostname }} {{ hostvars[host].ansible_fqdn }}:31360 check
+{% endfor %}
+    # server  s1 k8-w01.cyverse.at:31360 check # worker nodes hostname
+    # server  s2 k8-w02.cyverse.at:31360 check
+    # server  s3 k8-w03.cyverse.at:31360 check
+
+
+#---------------------------------------------------------------------
+# round robin balancing between the userportal backends
+#---------------------------------------------------------------------
+backend userportal
+    #redirect scheme https if !{ ssl_fc }
+    balance roundrobin
+    option  log-health-checks
+    option  forwardfor
+    # TODO server sX
+{% for host in groups['k8s-workers'] %}
+    server   {{ hostvars[host].ansible_hostname }} {{ hostvars[host].ansible_fqdn }}:30392 check
+{% endfor %}
+    # server  s1 k8-w01.cyverse.at:30392 check
```

Comment:
you remove these hosts

Reply:
This was to remind myself to change the hostname to `s<INDEX>`.
is done.